moveit_kinematics/cached_ik_kinematics_plugin/CMakeLists.txt

```diff
@@ -4,56 +4,63 @@ find_package(ur_kinematics QUIET)
 find_package(Boost COMPONENTS filesystem program_options REQUIRED)
 
 set(MOVEIT_LIB_NAME moveit_cached_ik_kinematics_base)
-add_library(${MOVEIT_LIB_NAME} src/ik_cache.cpp)
+add_library(${MOVEIT_LIB_NAME} SHARED src/ik_cache.cpp)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 target_link_libraries(${MOVEIT_LIB_NAME}
-    ${Boost_FILESYSTEM_LIBRARY}
-    ${catkin_LIBRARIES})
+    ${Boost_FILESYSTEM_LIBRARY})
+ament_target_dependencies(${MOVEIT_LIB_NAME}
+  rclcpp
+  moveit_core
+  moveit_msgs
+)
```

Comment:
You could also remove boost from `target_link_libraries` and add it to `ament_target_dependencies` as `Boost`

Reply:
yeah, we should use `ament_target_dependencies` wherever possible.

Reply:
done